update-readme.js

```diff
@@ -0,0 +1,156 @@
+
+// Category definitions for tools
+const categories = {
+  'Snapshot-based Interactions': [
+    ...snapshotTools.default,
+  ],
+  'Vision-based Interactions': [
+    ...screenTools.default
+  ],
+  'Tab Management': [
+   ...tabsTools.default(true),
+  ],
+  'Navigation': [
+    ...navigateTools.default(true),
+  ],
+  'Keyboard': [
+    ...keyboardTools.default(true)
+  ],
+  'Console': [
+    ...consoleTools.default
+  ],
+  'Files and Media': [
+    ...filesTools.default(true),
+    ...pdfTools.default
+  ],
+  'Utilities': [
+    ...commonTools.default(true),
+    ...installTools.default,
+    ...dialogsTools.default(true),
+  ],
+};
+
+/**
+ * Format tool information for README display
+ * @param {Object} tool - Tool information
+ * @returns {string} - Formatted markdown string
+ */
+function formatToolForReadme(tool) {
+  let markdown = `- **${tool.name}**\n`;
+  markdown += `  - Description: ${tool.description}\n`;
+  
+  if (tool.parameters && tool.parameters.length > 0) {
+    markdown += `  - Parameters:\n`;
+    tool.parameters.forEach(param => {
+      const meta = [];
+      if (param.type) meta.push(param.type);
+      if (param.optional) meta.push('optional');
+      
+      markdown += `    - \`${param.name}\` ${meta.length ? `(${meta.join(', ')})` : ''}: ${param.description}\n`;
+    });
+  } else {
+    markdown += `  - Parameters: None\n`;
+  }
+  
+  markdown += '\n';
+  return markdown;
+}
+
+/**
+ * Extract tool information from the schema
+ * @param {Object} schema - Tool schema
+ * @returns {Object} - Processed tool information
+ */
+function processToolSchema(schema) {
+  // Extract the input schema properties
+  const inputSchema = schema.inputSchema?.properties || {};
+  
+  // In JSON Schema, properties are considered optional unless listed in the required array
+  const requiredParams = schema.inputSchema?.required || [];
+  
+  const parameters = Object.entries(inputSchema).map(([name, prop]) => {
+    return {
+      name,
+      description: prop.description || '',
+      optional: !requiredParams.includes(name),
+      type: prop.type,
+    };
+  });
+
+  return {
+    name: schema.name,
+    description: schema.description,
+    parameters
+  };
+}
+
+/**
+ * Main function to update README
+ */
+async function updateReadme() {
+  console.log('Loading tool information from compiled modules...');
+  
+  // Read the README file
+  const readmePath = path.join(__dirname, 'README.md');
+  
+  // Generate new content for each category
+  let newReadmeContent = await fs.readFile(readmePath, 'utf-8');
+  
+  for (const [category, categoryTools] of Object.entries(categories)) {
+    if (categoryTools.length === 0) continue;
+    
+    const sectionHeader = `### ${category}`;
+    const startMarker = newReadmeContent.indexOf(sectionHeader);
+    
+    // If section exists
+    if (startMarker !== -1) {
+      const nextSectionMarker = newReadmeContent.indexOf('###', startMarker + sectionHeader.length);
```

Comment:
I'd not bother with replacing each section individually. Just replace the entire part with the tools descriptions since you generate all of them anyway. You could insert a magic comment for detecting the block of README that should be replaced and replace it in bulk, it would also  automatically work for newly added sections.

Reply:
I'd also add an HTML comment to mark the start and the end of the section explicitly with "This has been generated"

Reply:
Probably makes sense to add such a comment next to each tool too, so that contributors don't attempt to add stuff there.

Reply:
Vibe coded! ™️